src/loaders/ccloudResourceLoader.ts

```diff
@@ -283,14 +284,29 @@ export class CCloudResourceLoader extends CachingResourceLoader<
 
   /**
    * Query the Flink artifacts for the given CCloudFlinkDbKafkaCluster's CCloud environment + provider-region.
+   * Looks to the resource manager cache first, and only does a deep fetch if not found (or told to force refresh).
    * @returns The Flink artifacts for the given environment + provider-region.
    * @param resource The CCloud Flink Database (a Flink-enabled Kafka Cluster) to get the Flink artifacts for.
    */
-  public async getFlinkArtifacts(resource: CCloudFlinkDbKafkaCluster): Promise<FlinkArtifact[]> {
+  public async getFlinkArtifacts(
+    resource: CCloudFlinkDbKafkaCluster,
+    forceDeepRefresh: boolean,
```

Comment:
Why no default value set here like we do for [coarse resource loading](https://github.com/confluentinc/vscode/blob/fe7250fc5db7fd84b7682e3a57769195e1356911/src/loaders/cachingResourceLoader.ts#L67)?
```suggestion
    forceDeepRefresh: boolean = false,
```

Reply:
To force all the callers to know what mode they want. Is only the single caller, ... .